src/normalizers/keywordProtector.js

```diff
@@ -0,0 +1,49 @@
+import { regexMatchingDecoderMaker } from './regex.js';
+
+export default class KeywordProtector {
```

Comment:
maybe a closure instead of a class is enough for this? just a factory function that returns the 2 functions.

Reply:
Agree.  Changed.